lib/ui/components/form_components/styled_points_picker.widget.dart

```diff
@@ -0,0 +1,124 @@
+  final ValueChanged<int?> onItemSelected;
+  const StyledPointsPickerWidget({
+    required this.onItemSelected,
+    super.key,
+    this.pointsPickerdata,
+  });
+
+  @override
+  State<StyledPointsPickerWidget> createState() =>
+      _StyledPointsPickerWidgetState();
+}
+
+class _StyledPointsPickerWidgetState extends State<StyledPointsPickerWidget>
+    with SingleTickerProviderStateMixin {
+  late ValueNotifier<int?> _selectedIndexNotifier;
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedIndexNotifier = ValueNotifier<int?>(null);
+  }
+
+  @override
+  void dispose() {
+    _selectedIndexNotifier.dispose();
+    super.dispose();
+  }
+
+  void _selectItem(int index) {
+    _selectedIndexNotifier.value = index;
+    widget.onItemSelected(index);
+  }
+
+  List<CircularButton> get options =>
+      widget.pointsPickerdata?.pickerOptions ?? [];
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        if (widget.pointsPickerdata?.topLabel != null)
+          Padding(
+            padding: const EdgeInsets.only(left: AppSpacing.m),
+            child: Align(
+              alignment: widget.pointsPickerdata!.topLabelAlignment ??
+                  Alignment.centerLeft,
+              child: V2StyledTextWidget(
+                styledText: widget.pointsPickerdata!.topLabel!,
+              ),
+            ),
+          ),
+        Padding(
+          padding: const EdgeInsets.all(AppRadius.m),
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              ValueListenableBuilder<int?>(
+                valueListenable: _selectedIndexNotifier,
+                builder: (context, selectedIndex, child) {
+                  return Wrap(
+                    spacing: 12.0,
+                    runSpacing: 16.0,
+                    alignment: WrapAlignment.center,
+                    children: List.generate(options.length, (index) {
+                      final pickerOption = options[index];
+                      bool isSelected = selectedIndex == index;
+                      return getPickerOptions(
+                          pickerOption, index, isSelected, _selectItem);
+                    }),
+                  );
+                },
+              ),
+            ],
+          ),
+        ),
+        if (widget.pointsPickerdata?.bottomLabel != null)
+          Padding(
+            padding: const EdgeInsets.only(right: AppSpacing.m),
+            child: Align(
+              alignment: widget.pointsPickerdata!.bottomLabelAlignment ??
+                  Alignment.center,
+              child: V2StyledTextWidget(
+                  styledText: widget.pointsPickerdata!.bottomLabel!),
+            ),
+          ),
+      ],
+    );
+  }
+}
+
+Widget getPickerOptions(
+  CircularButton pickerOption,
+  int index,
+  bool isSelected,
+  ValueChanged<int> onItemSelected,
```

Comment:
```suggestion
  CircularButton pickerOption,
  int index,
  bool isSelected,
  ValueChanged<int> onItemSelected,
```

convert to named params